### PR TITLE
feat: add cpp demo for wechatqrcode

### PR DESCRIPTION
**Pull request by souhhmm**
Added a C++ demo for WeChatQRCode as per #135.
Files added/changed: `demo.cpp`, `CMakeLists.txt` and `README.md`.
**Edit:** Following the convention, I've refactored the code to be class based.

### Build:
image

### Using camera input:
image

### Using an image:
image

### CODE REVIEW

File: models/qrcode_wechatqrcode/demo.cpp [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+
+        if (!input_path.empty()) {
+            // process image
+            cv::Mat image = cv::imread(input_path);
+            if (image.empty()) {
+                std::cerr << "Could not read the image" << std::endl;
+                return -1;
+            }
+
+            auto [results, points] = qrDetector.detect(image);
```
Comment (fengyuentau) [inline models/qrcode_wechatqrcode/demo.cpp:130]: This is a C++17 standard feature IIRC, but OpenCV 4.x is based on C++11.

File: models/qrcode_wechatqrcode/demo.cpp [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+            cv::TickMeter tm;
+
+            while (true) {
+                cap >> frame;
+                if (frame.empty()) {
+                    std::cout << "No frames grabbed" << std::endl;
+                    break;
+                }
+
+                auto [results, points] = qrDetector.detect(frame);
```
Comment (fengyuentau) [inline models/qrcode_wechatqrcode/demo.cpp:164]: Same as above.

**Review by fengyuentau — COMMENTED** [review]
Thank you for contribution 👍 Please take a look at my comments below.

**Comment by souhhmm** [comment]
Hey, thanks for the feedback, I have updated the code.

File: models/qrcode_wechatqrcode/README.md [hunk 3]
```diff
@@ -21,6 +23,23 @@ python demo.py --input /path/to/image -v
 python demo.py --help
 ```
 
+### C++
+
+Install latest OpenCV and CMake >= 3.24.0 to get started with:
```
Comment (fengyuentau) [inline models/qrcode_wechatqrcode/README.md:28]: > Install latest OpenCV

Install latest OpenCV (with opencv_contrib).

**Review by fengyuentau — COMMENTED** [review]
I tried your demo locally on my host. It works great! Thank you👍

Before we merge this PR, could you add an extra information as below?

**Review by fengyuentau — APPROVED** [review]
👍

**Pull request merged**